debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresConnection.java

```diff
@@ -111,14 +116,20 @@ public PostgresConnection(JdbcConfiguration config, PostgresValueConverterBuilde
         }
     }
 
+    public PostgresConnection(JdbcConfiguration config, PostgresValueConverterBuilder valueConverterBuilder, String connectionUsage) {
+        this(config, valueConverterBuilder, connectionUsage,
+             config.getHostname().contains(":") ? JdbcConnection.patternBasedFactory(MULTI_HOST_URL_PATTERN, com.yugabyte.Driver.class.getName(), PostgresConnection.class.getClassLoader(), JdbcConfiguration.PORT.withDefault(PostgresConnectorConfig.PORT.defaultValueAsString()))
+                                                    : JdbcConnection.patternBasedFactory(URL_PATTERN, com.yugabyte.Driver.class.getName(), PostgresConnection.class.getClassLoader(), JdbcConfiguration.PORT.withDefault(PostgresConnectorConfig.PORT.defaultValueAsString())));
+    }
+
     /**
      * Create a Postgres connection using the supplied configuration and {@link TypeRegistry}
      * @param config {@link Configuration} instance, may not be null.
      * @param typeRegistry an existing/already-primed {@link TypeRegistry} instance
      * @param connectionUsage a symbolic name of the connection to be tracked in monitoring tools
      */
-    public PostgresConnection(PostgresConnectorConfig config, TypeRegistry typeRegistry, String connectionUsage) {
-        super(addDefaultSettings(config.getJdbcConfig(), connectionUsage), FACTORY, PostgresConnection::validateServerVersion, "\"", "\"");
+    public PostgresConnection(PostgresConnectorConfig config, TypeRegistry typeRegistry, String connectionUsage, ConnectionFactory factory) {
```

Comment:
This form of the constructor (with the extra factory parameter) is not really necessary